pkg/reconciler/taskrun/validate_resources.go

```diff
@@ -307,7 +307,7 @@ func validateTaskRunResults(tr *v1beta1.TaskRun, resolvedTaskSpec *v1beta1.TaskS
 			s = append(s, fmt.Sprintf(" \"%v\": %v", k, v))
 		}
 		sort.Strings(s)
-		return fmt.Errorf("mismatched types: %v", strings.Join(s, ","))
+		return fmt.Errorf("mismatched types, please check and make sure the results are written in valid JSON escaped format or the result type in taskSpec is correct: %v", strings.Join(s, ","))
```

Comment:
any way to make this a bit more concise?

Reply:
you mean divide the "invalida JSON" and "task not defined"?

Reply:
I revisit the code and think you're right. I added a hold to this PR. 
We should try to check them separately as @pritidesai suggested, but I haven't implemented the "Data Validation" yet

Reply:
Maybe once we could check the JSON data, we don't need this PR.

Reply:
It would be nice to check "invalid json" and "task results not defined" separately, but I think this is OK to merge if you can just condense it a bit, e.g. "Provided results don't match declared results; may be invalid JSON or missing result declaration"

Reply:
sure! Thanks for your suggestion!